### PR TITLE
refactor: columns regarding proving evolution

### PR DESCRIPTION
**Pull request by Freyskeyd**
# Description

This PR is refactoring a bit how certifier tasks and proving state evolution are managed.

## PR Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/agglayer-storage/src/columns/latest_proven_certificate_per_network/tests.rs
Comment (iljakuklic) [inline crates/agglayer-storage/src/columns/latest_proven_certificate_per_network/tests.rs:1]: Empty file.

File: crates/agglayer-certificate-orchestrator/src/lib.rs [hunk 1]
```diff
@@ -405,17 +408,104 @@ where
...
+                                "Failed to update the certificate header status: {:?}",
+                                error
+                            );
+                        }
+                    }
+
+                    // - 2. If the state knows the network and the height is the next one, we update
+                    //   the sate. This is the next certificate for this network.
+                    Entry::Occupied(mut entry) if *entry.get() + 1 == height => {
+                        entry.insert(height);
```
Comment (iljakuklic) [inline crates/agglayer-certificate-orchestrator/src/lib.rs:480]: Looks like there is some code duplication between the two branches:
* `Entry::Vacant(entry) if height == 0`
* `Entry::Occupied(mut entry) if *entry.get() + 1 == height`

File: crates/agglayer-certificate-orchestrator/src/lib.rs [hunk 2]
```diff
@@ -505,23 +617,24 @@ where
...
-                                            CertificateStatus::Pending,
-                                        )
-                                        .expect("Failed to insert certificate header");
+                                    // self.state_store
+                                    //     .insert_certificate_header(
+                                    //         &certificate,
+                                    //         CertificateStatus::Pending,
+                                    //     )
+                                    //     .expect("Failed to insert certificate
+                                    // header");
```
Comment (iljakuklic) [inline crates/agglayer-certificate-orchestrator/src/lib.rs:631]: Why is this commented out? Same a bit down around line 710.

File: crates/agglayer-certificate-orchestrator/src/lib.rs (hunk 1 above)
Comment (Freyskeyd) [reply crates/agglayer-certificate-orchestrator/src/lib.rs:480]: Yes, it is going to be refactored in another PR.

File: crates/agglayer-certificate-orchestrator/src/lib.rs [hunk 3]
```diff
@@ -405,17 +408,104 @@ where
...
+                        ) {
+                            error!(
+                                "Failed to update the certificate header status: {:?}",
+                                error
+                            );
+                        }
+                    }
+
+                    // - 2. If the state knows the network and the height is the next one, we update
+                    //   the sate. This is the next certificate for this network.
```
Comment (hadjiszs) [inline crates/agglayer-certificate-orchestrator/src/lib.rs:478]:
```suggestion
                    //   the state. This is the next certificate for this network.
```

**Review by hadjiszs — APPROVED** [review]

**Review by iljakuklic — APPROVED** [review]

**Pull request merged**